src/logger/exception_handler.cpp

```diff
@@ -22,13 +22,17 @@ namespace big
             exception_code == DBG_PRINTEXCEPTION_WIDE_C)
             return EXCEPTION_CONTINUE_SEARCH;
 
-        LOG(FATAL) << stack_trace(exception_info);
+        auto stack_trace_instance = std::make_unique<stack_trace>(exception_info);
+        LOG(FATAL) << *stack_trace_instance.get();
```

Comment:
No need to dereference the pointer, you can just modify or add a new stream function that takes a pointer as an argument.

```cpp
inline std::ostream& operator<< (std::ostream& os, const stack_trace* st)
{
    os << st.str();

    return os;
}
```

Reply:
I am used to create any class using std::make_uniptr, that will make me feel safe

Reply:
Allocating on heap is more expensive, try to allocate on stack as that'll be better (aka. how it was before).
If it causes side effects (like infinitely crashing or something), then you can use this method.

Reply:
sure